In [1]:
import pandas as pd
import plotly as py
import cufflinks as cf

cf.set_config_file(offline=True, theme='ggplot')

# 人口密度

In [2]:
df = pd.read_csv("人口密度.csv",encoding='gbk')
df[['地区','人口密度']].head()

地区        人口密度
0    北京  838.391225
1    天津  905.432781
2   石家庄  619.384149
3    太原  538.248321
4  呼和浩特  142.736879

In [3]:
df['人口密度'].describe()

count      36.000000
mean      697.749144
std       516.344275
min        18.781760
25%       361.142947
50%       600.689035
75%       923.156006
max      2306.593060
Name: 人口密度, dtype: float64

In [4]:
from pyecharts.faker import Faker
from pyecharts import options as opts
from pyecharts.charts import Geo
from pyecharts.globals import ChartType, SymbolType


def geo_base() -> Geo:
    c = (
        Geo()
        .add_schema(maptype="china",itemstyle_opts=opts.ItemStyleOpts(color="#323c48", border_color="#111"))
        .add("人/平方千米", zip(list(df.地区),list(df.人口密度)))
        .set_series_opts(label_opts=opts.LabelOpts(is_show=False))
        .set_global_opts(
            title_opts=opts.TitleOpts(title="2018年36个主要城市的人口密度情况"),
            visualmap_opts=opts.VisualMapOpts(max_=2306.6,min_=18.7,is_piecewise=True,
                                             pieces=[{'min':924,'max':2306.593060},{'min':602,'max':923.156006},{'min':362,'max':600.689035},{'min':18.781760,'max':361.142947}]))
    )
    return c

In [5]:
geo_base().render_notebook()

人口密度越大，代表该地人口越多。人口越多通常意味着该地的人口职业竞争可能会相对较激烈，那么该地的劳动力人口的压力可能也会随之较大。通过按照25%的比例递增，将中国36个主要城市（大中城市）的人口密度分为了四个层次。其中，一线城市上海、广州和深圳的人口密度不出所料地排在了前75%（前9位），但北京却位于第11名,那是否意味着北京比其余三个一线城市失业率更高、工资水平更低、消费压力更大等而导致人口流失？同时在前75%中，除一线城市和厦门外，剩余的郑州、南京、武汉、成都、西安都属于今年的新一线城市，可以发现，新一线城市成为了很多人生活和奋斗上的选择。

前五名：上海、深圳、厦门、广州、郑州  
后五名：银川、哈尔滨、乌鲁木齐、呼和浩特、拉萨

# 人均消费支出与平均薪资

In [6]:
df2 = pd.read_csv("人均消费支出与平均薪资.csv",encoding='gbk')
df2.head()

地区   平均薪资  居民人均消费支出       差值
0  呼和浩特  71387   22714.0  48673.0
1    厦门  85166   33192.0  51974.0
2   哈尔滨  71771   19221.0  52550.0
3    海口  77632   22971.0  54661.0
4  乌鲁木齐  85990   28404.0  57586.0

In [7]:
df2.describe()

平均薪资      居民人均消费支出             差值
count      36.000000     34.000000      34.000000
mean    92374.583333  25079.117647   67914.411765
std     18077.509741   7648.599522   14663.338822
min     71387.000000  11520.000000   48673.000000
25%     81991.250000  19580.250000   60259.750000
50%     86557.500000  22394.000000   64556.500000
75%     95551.000000  29644.000000   69517.750000
max    149843.000000  43351.000000  115416.000000

In [8]:
from pyecharts.charts import Line

def line_smooth() -> Line:
    line = (
        Line()
        .add_xaxis(list(df2.地区))
        .add_yaxis("人均消费支出（元）", list(df2.居民人均消费支出))
        .add_yaxis("平均工资（元）", list(df2.平均薪资))
        .set_global_opts(title_opts=opts.TitleOpts(title="平均工资和消费支出的关系"),
            datazoom_opts=[opts.DataZoomOpts(), opts.DataZoomOpts(type_="inside")],
        )
    )
    return line

In [9]:
line_smooth().render_notebook()

由图中可看出，两条折线的运动趋势相似，由此可以判断出消费支出与薪酬呈显性的正比关系，也就是说薪酬越高的城市，其消费支出也会相应越高。其中，有趣的发现是，位于偏远地区的拉萨，人们平均工资达到每年12.6936万元,仅次于北京和上海，位居第3。并且，从已知的平均工资的数据来看，拉萨的人均消费支出在36个主要城市中是最低的。由此可以判断出，在36个主要城市中，拉萨居民在财力上的压力是最小的。

In [14]:
from pyecharts import options as opts
from pyecharts.charts import Bar


def chazhi() -> Bar:
    c = (
        Bar()
        .add_xaxis(list(df2.地区))
        .add_yaxis("差值", list(df2.差值))
        .set_global_opts(title_opts=opts.TitleOpts(title="差值", subtitle="平均工资-人均消费支出"),
                        datazoom_opts=[opts.DataZoomOpts(), opts.DataZoomOpts(type_="inside")],)
    )
    return c

In [15]:
chazhi().render_notebook()

In [10]:
dfa=df2.set_index('地区').loc[:,['差值']]

In [11]:
dfa.iplot(kind="bar", xTitle="地区",yTitle="差值", title="差值" )

由平均工资和消费支出关系图可以看出，人均消费支出中最突出的是北京、上海和深圳，基本达到4万左右。那么从平均工资减去人均消费支出后所剩的储蓄即差值来看，在36个城市中北京、上海的人均的基本储蓄是比较高的，基本可以达到10万左右，而相对北京和上海来说的深圳人均的基本储蓄比较低，为7万1千多。由此大致可知，在一线城市中，深圳的居民消费压力是最大的。  

并且从差值图中也能直观地看出，在36个城市中，呼和浩特的居民消费压力是最大的。36个主要城市中人均消费支出的平均值是25079，呼和浩特的人均消费支出是低于平均值的，那么呼和浩特人均的基本储蓄最低主要原因在于当地居民的平均工资比较低，其平均工资是36个主要城市中的最低的，呼和浩特人口密度小原因之一或许也正是因为呼和浩特低工资。


前五名：拉萨、北京、上海、广州、南京  
后五名：呼和浩特、厦门、哈尔滨、海口、乌鲁木齐

# 住宅商品房平均房价 

In [115]:
df3 = pd.read_csv("住宅商品房平均房价.csv",encoding='gbk')
df3.head()

地区     2018年    2017年    2016年    2015年    2014年    2013年     2012年  \
0  深圳  55441.01  48622.0  45498.0  33661.0  24040.0  23427.0  18995.92   
1  北京  37420.19  34117.0  28489.0  22300.0  18499.0  17854.0  16553.48   
2  上海  28981.11  24866.0  25910.0  21501.0  16415.0  16192.0  13869.88   
3  厦门  33715.43  28053.0  25251.0  18928.0  17778.0  14551.0  12953.38   
4  南京  19708.33  15259.0  17884.0  11260.0  10964.0  11078.0   9674.84   

      2011年    2010年    2009年  
0  21037.05  18954.0  14389.0  
1  15517.90  17151.0  13224.0  
2  13565.83  14290.0  12364.0  
3  13422.59  11590.0   8935.0  
4   8414.62   9227.0   6893.0

In [116]:
df3.describe()

2018年         2017年         2016年         2015年         2014年  \
count     35.000000     35.000000     35.000000     35.000000     35.000000   
mean   14411.196000  12414.600000  11034.485714   9519.971429   8598.742857   
std    10428.099736   9132.826549   8544.984189   6151.713350   4651.173169   
min     5589.490000   4892.000000   4448.000000   4498.000000   4111.000000   
25%     8220.610000   7493.500000   6250.000000   6181.500000   5802.500000   
50%    10840.490000   8827.000000   7707.000000   7303.000000   6579.000000   
75%    15152.760000  12919.500000  11398.000000  10476.500000   9513.000000   
max    55441.010000  48622.000000  45498.000000  33661.000000  24040.000000   

              2013年         2012年         2011年         2010年         2009年  
count     35.000000     35.000000     35.000000     35.000000     35.000000  
mean    8372.685714   7718.497714   7439.300000   7059.800000   5705.971429  
std     4424.655499   3709.090984   3903.408026   4047.242277   3055.166911  
min     4380.000000   4187.260000   3438.600000   3196.000000   2811.000000  
25%     5744.000000   5339.125000   4831.035000   4293.500000   3611.500000  
50%     6639.000000   6224.030000   5969.660000   5502.000000   4463.000000  
75%     9272.500000   8842.210000   8481.130000   8004.500000   6523.000000  
max    23427.000000  18995.920000  21037.050000  18954.000000  14389.000000

In [117]:
from pyecharts import options as opts
from pyecharts.charts import Bar, Page, Pie, Timeline

def timeline2() -> Timeline:
    x = list(df3.地区)
    t2 = Timeline()
    for i in range(2009, 2019):
        c = (
        Line()
        .add_xaxis(x)
        .add_yaxis("住宅商品房平均房价", list(df3["{}年".format(i)]))
        .set_global_opts(title_opts=opts.TitleOpts("{}年各城市住宅商品房平均房价".format(i)),
                             datazoom_opts=opts.DataZoomOpts(type_="inside"),
                             visualmap_opts=opts.VisualMapOpts(max_=56000,min_=2811),
                             yaxis_opts=opts.AxisOpts(max_=56000,min_=2811))
        )
        
        t2.add(c, "{}年".format(i),)
    return t2


In [118]:
timeline2().render_notebook()

现代结婚最重要的设施就是房子，也是现代年轻人压力的重要来源。那么近10年来36个主要城市的房价走势和未来的趋势会怎样呢？由图表可知，前7年大多数城市的房价上升趋势比较缓和。但自2016开始大多数地区房价出现暴涨，主要在于2015年底国务院出台去库存政策刺激房地产，房价就在深圳等一线城市出现疯涨的行情，楼市“四小龙”南京、合肥、厦门、苏州接棒领衔，其它一、二线热点城市纷纷轮番上涨。可以发现2018年房价上涨稍微放缓，预测未来几年房价依旧会上涨，但是涨幅会缩小。  
那么近10年房价涨幅比较低的城市有银川、南宁、兰州、西宁、长春。

In [136]:
from pyecharts.charts import Line

def line_smooth2() -> Line:
    line = (
        Bar()
        .add_xaxis(list(df2.地区))
        .add_yaxis("2009年", list(df3['2009年']))
        .add_yaxis("2010年", list(df3['2010年']))
        .add_yaxis("2011年", list(df3['2011年']))
        .set_global_opts(title_opts=opts.TitleOpts(title="2009-2011年各城市住宅商品房平均房价变化"),
                         legend_opts=opts.LegendOpts(pos_left=380),
            datazoom_opts=[opts.DataZoomOpts(), opts.DataZoomOpts(type_="inside")],
        )
    )
    return line

In [137]:
line_smooth2().render_notebook()

In [134]:
from pyecharts.charts import Line

def line_smooth3() -> Line:
    line = (
        Bar()
        .add_xaxis(list(df2.地区))
        .add_yaxis("2012年", list(df3['2012年']))
        .add_yaxis("2013年", list(df3['2013年']))
        .add_yaxis("2014年", list(df3['2014年']))
        .set_global_opts(title_opts=opts.TitleOpts(title="2012-2014年各城市住宅商品房平均房价变化"),
                         legend_opts=opts.LegendOpts(pos_left=380),
            datazoom_opts=[opts.DataZoomOpts(), opts.DataZoomOpts(type_="inside")],
        )
    )
    return line

In [135]:
line_smooth3().render_notebook()

In [132]:
from pyecharts.charts import Line

def line_smooth4() -> Line:
    line = (
        Bar()
        .add_xaxis(list(df2.地区))
        .add_yaxis("2015年", list(df3['2015年']))
        .add_yaxis("2016年", list(df3['2016年']))
        .add_yaxis("2017年", list(df3['2017年']))
        .add_yaxis("2018年", list(df3['2018年']))
        .set_global_opts(title_opts=opts.TitleOpts(title="2015-2018年各城市住宅商品房平均房价变化"),
                         legend_opts=opts.LegendOpts(pos_left=380),
                         datazoom_opts=[opts.DataZoomOpts(), opts.DataZoomOpts(type_="inside")],
        )
    )
    return line

In [133]:
line_smooth4().render_notebook()

# 失业率

In [138]:
df4 = pd.read_csv("失业率.csv",encoding='gbk')
df4.head()

地区  2018年  2017年  2016年  2015年  2014年
0   哈尔滨   3.76   3.68   3.76   3.88   3.72
1  呼和浩特   3.60   3.65   3.66   3.56   3.50
2    银川   3.58   3.61   3.92   4.05   4.09
3    上海   3.57   3.90   4.10   4.10   4.20
4   石家庄   3.50   3.90   4.10   4.00   4.10

In [139]:
df4.describe()

2018年      2017年      2016年      2015年      2014年
count  36.000000  36.000000  36.000000  36.000000  36.000000
mean    2.727500   2.819722   2.870556   2.834444   2.843889
std     0.698531   0.702524   0.776674   0.810056   0.781710
min     1.340000   1.300000   0.900000   0.900000   1.310000
25%     2.185000   2.350000   2.297500   2.200000   2.260000
50%     2.940000   2.930000   3.015000   2.990000   2.965000
75%     3.347500   3.402500   3.470000   3.447500   3.425000
max     3.760000   3.900000   4.100000   4.100000   4.200000

In [140]:
from pyecharts import options as opts
from pyecharts.charts import Bar, Page, Pie, Timeline


def timeline2() -> Timeline:
    x = list(df4.地区)
    tl = Timeline()
    for i in range(2014, 2019):
        bar = (
            Bar()
            .add_xaxis(x)
            .add_yaxis("失业率", list(df4["{}年".format(i)]))
            .set_global_opts(title_opts=opts.TitleOpts("{}年各城市失业率".format(i)),
                             datazoom_opts=(opts.DataZoomOpts(type_="inside")),
                             visualmap_opts=opts.VisualMapOpts(max_=4.5,min_=0),
                            yaxis_opts=opts.AxisOpts(max_=4.5,min_=0)))
        tl.add(bar, "{}年".format(i),)
    return tl

In [141]:
timeline2().render_notebook()

失业率和就业率存在反比关系，失业低则意味着就业率高，同时也意味着就业竞争压力小。在国家积极的就业政策背景下，近5年来大多数城市的失业率稳中有降，但仍有些城市的失业率始终在上升或者失业率始终维持在较高水平。例如成都近5年的失业率一直在上升；哈尔滨的失业率始终保持一年升一年降的趋势，但变化起伏不大，近5年来失业率始终保持在36个主要城市中的前5名。并且东北以及靠近东北的北方地区的失业率大多数都非常高，近5年的失业率基本都高于36个城市的平均失业率。而近5年失业率始终保持较低水平且无明显上升趋势的城市主要有南京、杭州、宁波、海口以及除深圳外的一线城市等。

## 总结（36个主要城市六个维度排名总和）

In [145]:
df5 = pd.read_csv("生活压力排名.csv",encoding='gbk')
df5[['地区','排名总和']].head()

地区  排名总和
0   厦门    41
1   上海    47
2   天津    65
3   深圳    65
4  石家庄    77

In [146]:
from pyecharts.faker import Faker
from pyecharts import options as opts
from pyecharts.charts import Funnel, Page

def funnel_label_inside() -> Funnel:
    c = (
        Funnel()
        .add(
            "排名总和",
            [list(z) for z in zip(df5['地区'], df5['排名总和'])],
            label_opts=opts.LabelOpts(position="inside"),
        )
        .set_global_opts(title_opts=opts.TitleOpts(title="36个主要城市六个维度排名总和"),
                        legend_opts=opts.LegendOpts(type_='scroll',pos_top=25)
                        )
    )
    return c

In [147]:
funnel_label_inside().render_notebook()

本项目从人口密度、人均消费支出、平均薪资、住宅商品房房价和失业率六个维度探索了我国36个主要城市的生活压力。通过对2018年36个城市的六个维度进行了排名的总和计算，排名总和与生活压力成反比关系，即总和越高的城市，其居民的生活压力越低。在大多数人都渴望能够立足的一线城市中，除了失业率外，一线城市的每一维度基本都位于前列。那么从排名总和来看，一线城市中生活压力较低的是北京，最高的是上海。除住宅商品房房价外，北京在其余方面基本都优于其他一线城市。那么与人口流入收紧、就业竞争激烈的一线城市相比，其余的32个主要城市的排名总和又是怎样的呢？在二线城市中，兰州的生活压力是最低的，除平均薪资外，其余5个方面都位居前8名。而作为非一线城市和新一线城市的厦门的生活压力却是最高的，甚至是所有城市中最高的。虽然厦门生活压力如此大，但仍有许多人选择生活在厦门，或许厦门留有令人心动的方面？在三线城市中西宁的生活压力是最低的，海口是最高的（缺乏拉萨的完整数据）。